key.core/src/main/java/de/uka/ilkd/key/speclang/njml/Translator.java

```diff
@@ -2531,6 +2552,22 @@ protected void addWarning(ParserRuleContext node, String description) {
         exc.addWarning(description, node.start);
     }
 
+    private void warnPotentiallyUnintendedFramingSemantics(
+            ParserRuleContext ctx, String clauseName) {
+        clauseName =
+            clauseName.startsWith("loop_") ? clauseName.replaceFirst("loop_", "") : clauseName;
+        for (final String s : DISCOURAGED_CLAUSE_NAMES) {
+            if (clauseName != null && clauseName.startsWith(s)) {
+                addWarning(ctx, clauseName + " does not conform to KeY's supported JML language, "
```

Comment:
This message imo reads a bit as if we would support the "one and only" JML (in particular, the "other JML-like languages" part). I think the message would sound better if it talked about "JML dialects" or similar.

Reply:
I rewrote the error message accordingly.